pkg/apis/sources/v1alpha1/awssqs_lifecycle.go

```diff
@@ -67,3 +68,13 @@ func (s *AWSSQSSource) GetEventTypes() []string {
 func (s *AWSSQSSource) AsEventSource() string {
 	return s.Spec.ARN.String()
 }
+
+// WantsOwnServiceAccount implements serviceAccountProvider.
+func (*AWSSQSSource) WantsOwnServiceAccount() bool {
+	return false
+}
+
+// ServiceAccountOptions implements serviceAccountProvider.
+func (*AWSSQSSource) ServiceAccountOptions() []func(*corev1.ServiceAccount) {
+	return nil
+}
```

Comment:
Here is a dummy implementation of the interface which, for now, doesn't do anything.

In a subsequent PR, that source will
- return `WantsOwnServiceAccount -> true` when the user selects the "IAM role" authentication.
- return a ServiceAccountOptions which sets the `eks.amazonaws.com/role-arn` to the arn of the selected IAM role.